### DISCUSSION TITLE
Installing packages with colliding executable produces no error

### DISCUSSION THREAD

**Discussion by cdaringe** [pnpm/pnpm#15987] [Ideas]
### Verify latest release

- [x] I verified that the issue exists in the latest pnpm release

### pnpm version

8.6.12

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/cdaringe/pnpm-colliding-executable

### Reproduction steps

1. have two dependencies in your package that both install executables, e.g.

```json
{
  "dependencies": {
    "prettier-2": "npm:prettier@2",
    "prettier": "3.0.3"
  }
}
```

2. `pnpm install`
3. `pnpm exec prettier --version`

### Describe the Bug

Two packages are requested but cannot logically coexist, but were accepted anyway.

### Expected Behavior

An exception should have been thrown.

A _feature_ could be released to _support_ this impossible case as well

1. `.npmrc` - `unique-bin-names-on-collision=true`
  2. in the above example, then generated bin is prefixed with its package (and or package version), `prettier-2_prettier`
2. `package.json::pnpm::bin::alias::[packageName]::{ [oldBinName]: newBinName`
e.g.
```json
{
  "pnpm": {
     "bin": {
        "alias": {
           "prettier-2": {
              "prettier": "prettier2"
            }
        }
     }
  }
}
```
3. Similar to the above, but a feature more  like "install no bins" for `<packageName/packageVersion>`

Really, just an exception for starters would be great

### Which Node.js version are you using?

16.10.x

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by Shuuren** [comment]
I'm working on a fix for this.

**Comment by xshubhamg** [comment]
I'm working on a fix for this.

**Comment by Shuuren** [comment]
Stopping work on this — parked; feature/ideas shape, not a clean bug fix.
